Sources/SnapshotTesting/Strategy/CALayer.swift

```diff
@@ -28,16 +28,29 @@ import UIKit
 
 extension Strategy where Snapshottable == CALayer, Format == UIImage {
   public static var image: Strategy {
-    return .image(precision: 1)
+    return .image()
   }
 
-  public static func image(precision: Float) -> Strategy {
-    return SimpleStrategy.image(precision: precision).pullback { layer in
-      UIGraphicsImageRenderer(size: layer.bounds.size).image { context in
-        layer.setNeedsLayout()
-        layer.layoutIfNeeded()
-        layer.render(in: context.cgContext)
+  public static func image(precision: Float = 1, traits: UITraitCollection = .unspecified)
+    -> Strategy {
+      return SimpleStrategy.image(precision: precision).pullback { layer in
+        layer.image(for: traits)
       }
+  }
+}
+
+extension CALayer {
+  func image(for traits: UITraitCollection) -> UIImage {
+    let renderer: UIGraphicsImageRenderer
+    if #available(iOS 11.0, *) {
+      renderer = UIGraphicsImageRenderer(size: self.bounds.size, format: .init(for: traits))
```

Comment:
This allows the renderer to take things like gamut/scale from the trait collection to render a proper image size on simulators that may differ!

Reply:
Edit: I've removed the gamut/scale from the device trait collections for now, though people are still free to use them. I was seeing issues with layout and console warnings with CGColorSpace when loading images this way. We may be able to solve them in the future, but this PR still at least paves the way.